dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/dataelement/DataElementDeletionHandler.java

```diff
@@ -138,8 +124,7 @@ private void deleteLegendSet( LegendSet legendSet )
 
     private DeletionVeto allowDeleteOptionSet( OptionSet optionSet )
     {
-        String sql = "SELECT COUNT(*) FROM dataelement WHERE optionsetid = " + optionSet.getId();
-
-        return jdbcTemplate.queryForObject( sql, Integer.class ) == 0 ? ACCEPT : VETO;
+        String sql = "SELECT COUNT(*) FROM dataelement WHERE optionsetid = :id";
```

Comment:
Do we want to change to lower-case SQL while we are at it?

Reply:
:+1: I can make another PR for that